apps/sms/views/shared/test/unit/utils_test.js

```diff
@@ -1153,6 +1153,131 @@ suite('Utils', function() {
     });
   });
 
+  suite('Utils.throttle', function() {
+    test('right typeof parameters', function() {
+      assert.throw(() => Utils.throttle(null), 
+                   Error, 
+                   'func must be a Function');
+      assert.throw(() => Utils.throttle(() => {}, null),
+                   Error, 
+                   'delay must be a positive number');
+      assert.throw(() => Utils.throttle(() => {}, -1),
+                  Error, 
+                  'delay must be a positive number');
+    });
+
+    test('parameters are passed to the throttled func', function() {
+      var dummyObjectA = {};
+      var dummyObjectB = {};
+
+      var func = function() {
+        assert.equal(arguments[0], dummyObjectA);
+        assert.equal(arguments[1], dummyObjectB);
+      };
+
+      var throttlerFunc = Utils.throttle(func, 100);
+      
+      throttlerFunc(dummyObjectA, dummyObjectB);
+    });
+
+    test('testing this in the throttled func', function() {
+      var self = this;
+      var dummyObject = {};
+
+      var throttlerFunc = Utils.throttle(function() {
+        assert.equal(this, dummyObject);
+      }, 100);
+      throttlerFunc.call(dummyObject);
+
+      var throttlerFunc2 = Utils.throttle(()=>{
+        assert.equal(this, self);
+      }, 100);
+      throttlerFunc2.call(dummyObject);
+    });
+
+    test('preventing first call', function() {
+      var delay = 100;
+      var firstCallFuncToExecute = sinon.stub();
+      var firstCallThrottleFuncToExecute = 
+        Utils.throttle(firstCallFuncToExecute, delay);
+      
+      var notFirstCallFuncToExecute = sinon.stub();
+      var notFirstCallThrottleFuncToExecute = 
+      Utils.throttle(notFirstCallFuncToExecute, delay, true);
+
+      firstCallThrottleFuncToExecute();
+      sinon.assert.calledOnce(firstCallFuncToExecute);
+
+      notFirstCallThrottleFuncToExecute();
+      sinon.assert.notCalled(notFirstCallFuncToExecute);
+    });
+
+    test('preventing last call', function() {
+      var clock = sinon.useFakeTimers(Date.now());
```

Comment:
Actually 4 on 8 of the tests do not require fakeTimer to pass so I wondered if that was not a better idea to use fakeTimer only where it's needed and use real (none faked) Date and setTimeout objects when possible (which is closer to real usage).
Tell me if you don't agree and I'll move fakeTimer to the suite setup !